tests/misc/CachingUtilitiesTest.py

```diff
@@ -66,6 +67,11 @@ def test_settings_change(self):
         settings_hash = get_settings_hash(sections)
         self.assertTrue(settings_changed(self.log_printer, settings_hash))
 
+        with patch('coalib.misc.CachingUtilities.pickle_load') as \
+                mocked_pickle_load:
+            mocked_pickle_load.return_value = {}
+            self.assertFalse(settings_changed(self.log_printer, settings_hash))
```

Comment:
This does not solve the problem. It just patches the symptom.

To properly solve this, create a `tearDown` that deletes all side effects from the tests after the tests are done.

Reply:
By this, I mean to say remove all keys related to tests from the pickle after the tests are done.

Reply:
@harshhx17 replied at https://github.com/coala/coala/pull/5216#issuecomment-383381767

Reply:
Replying to ^ , this isnt about what you have done or not done.

@adtac is asking you to re-create the *scenario* which this `if` is supposed to address.

Your code does force the use of this `if` branch, which does achieve 100% coverage, but it doesnt reproduce the scenario, so the objective of 100% coverage isnt obtained.

The line `settings_hash_db = pickle_load(None, 'settings_hash_db', {})` will already return `{}`, so you dont need to mock `pickle_load`.  You could mock something quite a bit deeper to cause `{}` to be returned.  That may not make @adtac happy, but at least you are more closely recreating the scenario.

@adtac is saying you should use `setUp` and `tearDown` to create the scenario, e.g. create a backup of the setting cache, remove the settings for this directory, and then restore the settings again after the test is completed.

As it is an invasive test process, I suggest putting it into its own test class.